src/Bicep.Core/Semantics/FixedFunctionParameter.cs

```diff
@@ -21,10 +21,20 @@ public FixedFunctionParameter(string name, string description, TypeSymbol type,
 
         public TypeSymbol Type { get; }
 
-        public bool Required { get; }
+        public FunctionParameterFlags Flags { get; }
+
+        public bool Required => Flags.HasFlag(FunctionParameterFlags.Required);
 
         public string Signature => this.Required
             ? $"{this.Name}: {this.Type}"
             : $"[{this.Name}: {this.Type}]";
     }
+
+    public enum FunctionParameterFlags
```

Comment:
[nit] Move this to its own file, and ensure you use the `[Flags]` annotation on it.

Reply:
Removed and used TypeSymbolValidationFlags